lib/dbservice/users.ex

```diff
@@ -177,6 +177,38 @@ defmodule Dbservice.Users do
     Repo.get_by(Student, student_id: student_id)
   end
 
+  @doc """
+  Gets a student by either student_id or apaar_id.
+  Returns the first student found with either identifier.
+
+  ## Examples
+
+      iex> get_student_by_id_or_apaar_id(%{"student_id" => "1234", "apaar_id" => nil})
+      %Student{}
+      iex> get_student_by_id_or_apaar_id(%{"student_id" => nil, "apaar_id" => "123456789101"})
+      %Student{}
+      iex> get_student_by_id_or_apaar_id(%{"student_id" => nil, "apaar_id" => nil})
+      nil
+  """
+  def get_student_by_id_or_apaar_id(%{"student_id" => student_id, "apaar_id" => apaar_id}) do
+    cond do
+      student_id && student_id != "" ->
+        get_student_by_student_id(student_id)
+
+      apaar_id && apaar_id != "" ->
+        Repo.get_by(Student, apaar_id: apaar_id)
+
+      true ->
+        nil
+    end
+  end
+
+  def get_student_by_id_or_apaar_id(record) when is_map(record) do
+    student_id = Map.get(record, "student_id")
+    apaar_id = Map.get(record, "apaar_id")
+    get_student_by_id_or_apaar_id(%{"student_id" => student_id, "apaar_id" => apaar_id})
```

Comment:
I don't understand this function only :D Is this a recursive function? Why?

Reply:
No this is not a recursive function. This function calls a different function with the same name (it's calling the first clause of the function). This is function overloading/pattern matching.

When we already have a properly structured map like %{"student_id" => "123", "apaar_id" => "456"}, first clause handles this directly

When we have a raw record that might have different field names or structure
record = %{
  "student_id" => "123", 
  "apaar_id" => "456",
  "other_field" => "value",
  "..." => "..."
}

Second clause handles this by extracting only needed fields